File: front/src/app/shared/model/dto/evento-pessoa-dto.ts
import {StatusEnum} from "@/shared/model/enum/status.enum";
import {PessoaDto} from "@/shared/model/dto/pessoa-dto";

export class EventoPessoaDto {

    id?: number;
    pessoa?: PessoaDto;
    status?: StatusEnum;
    nomeMagicNumber?: string;
    jaEscolheu?: boolean;
    version?: number;

}

File: back/src/main/java/br/eng/eliseu/presente/model/dto/EventoPessoaDto.java
package br.eng.eliseu.presente.model.dto;

import br.eng.eliseu.presente.model.StatusEnum;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder(toBuilder = true)
public class EventoPessoaDto {

    private Long id;

    private PessoaDto pessoa;

    private StatusEnum status;

    private String nomeMagicNumber; // token

    // Somente leitura: indica se a pessoa já possui uma escolha ATIVA neste evento
    private Boolean jaEscolheu;

    private Long version;
}
